CHANGELOG.md

```diff
@@ -5,6 +5,24 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [v3.35.2](https://github.com/buildkite/agent/tree/v3.35.2) (2022-04-13)
+[Full Changelog](https://github.com/buildkite/agent/compare/v3.35.1...v3.35.2)
+
+### Fixed
+- Fix race condition in bootstrap.go [#1606](https://github.com/buildkite/agent/pull/1606) (@moskyb)
+
+### Changed
+- Bump some dependency versions - thanks @dependabot!
+  - github.com/stretchr/testify: 1.5.1 -> 1.7.1 [#1608](https://github.com/buildkite/agent/pull/1608)
+  - github.com/mitchellh/go-homedir: 1.0.0 -> 1.1.0 [#1576](https://github.com/buildkite/agent/pull/1576)
+
+## [v3.35.1](https://github.com/buildkite/agent/tree/v3.35.1) (2022-04-05)
+[Full Changelog](https://github.com/buildkite/agent/compare/v3.35.0...v3.35.1)
+
+### Fixed
+
+- Revert file permission changes made in [#1580](https://github.com/buildkite/agent/pull/1580). They were creating issues with docker-based workflows [#1601](https://github.com/buildkite/agent/pull/1601) (@pda + @moskyb)
```

Comment:
Nit: Maybe a capital D for Docker?

Reply:
i think i'll probably leave this as is for now - i'm just pulling in the changes from another branch, and it's worthwhile to keep things in sync between the two branches. our users will still be able to identify docker as a common noun, i hope :P